yeonjae/programmers-2/캐시.js

```diff
@@ -0,0 +1,40 @@
+function solution(cacheSize, cities) {
+  const HIT = 1;
+  const MISS = 5;
+  const cache = [];
+  let answer = 0;
+
+  if (cacheSize === 0) {
+    return cities.length * MISS;
+  }
+
+  while (cities.length) {
+    const city = cities.shift().toLowerCase();
+
+    if (cache.includes(city)) {
+      cache.splice(cache.indexOf(city), 1);
+      cache.push(city);
+      answer += HIT;
+    } else {
+      if (cache.length === cacheSize) {
+        cache.shift();
+      }
+
+      cache.push(city);
+      answer += MISS;
+    }
+  }
+  return answer;
+}
```

Comment:
```suggestion
function solution(cacheSize, cities) {
  const HIT = 1;
  const MISS = 5;
  const cache = [];
  let answer = 0;

  if (!cacheSize) return cities.length * MISS;

  cities.forEach((city) => {
    const loweredCity = city.toLowerCase();
    const cacheIdx = cache.indexOf(loweredCity);
      
    if (cacheIdx === -1) {
      if (cache.length === cacheSize) {
        cache.shift();
      }
      answer += MISS;
    } else {
      cache.splice(cacheIdx, 1);
      answer += HIT;
    }
    cache.push(loweredCity);
  });
    
  return answer;
}
```
- 풀이하느라 고생하셧씁니다.
- includes와 indexOf를 같이 사용하는 것은 불 필요한 코드입니다. indexOf를 통해 idx를 구하고, 이를 -1인지 아닌지로 포함하는지 체크하고 splice에서도 해당 idx를 넣어서 원하는 결과를 가져올 수 있습니다
- HIT, MISS 상수를 이용한 부분은 아주 좋습니다 👍
- while문 + shift를 효율성이 낮을 것 같아 forEach로 변경하였습니다